Read keyboardhook, gatewayaccesstoken and gatewaycredentialssource from RDP File

asherbiny opened the pull request:
## Description
Upon importing an RDP file mRemoteNG will read and apply the above 3 fields to the connection being imported.



## Motivation and Context
Currently there is a lot of friction when using Gateways with access tokens. If a user imports an RDP file which contains a Gateway access token, they have to set the dropdown menu to "Use RD Gateway access token" and then manually open the file (via a text editor) and copy/paste the access token into mRemoteNG's UI. This provides for a really poor user experience.
Issue: https://github.com/mRemoteNG/mRemoteNG/issues/2722)https://github.com/mRemoteNG/mRemoteNG/issues/2722

## How Has This Been Tested?
-Ran mRemoteNG in both Debug and Release modes. 
-Confirmed that Redirect Key Combinations, RDP-Gateway Access Token and RDP-Gateway Credentials are all reflective of the settings in the RDP file.
-Verified that key combinations are being correctly redirected with the setting ON 
-Verified that mRemoteNG is now able to connect to the Gateway using the embedded access token (without entering it manually)

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Changed feature (non-breaking change which changes functionality)
- [ ] Changed feature (**breaking** change which changes functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Updated translation

## Checklist:
- [X] I have read the **CONTRIBUTING** document.
- [X] My code follows the code style of this project.
- [ ] All Tests within VisualStudio are passing
- [X] This pull request does not target the master branch.
- [ ] I have updated the changelog file accordingly, if necessary.
- [ ] I have updated the documentation accordingly, if necessary.

asherbiny committed:
Read keyboardhook, gatewayaccesstoken and gatewaycredentialssource from RDP File

With this commit mRemoteNG, upon importing an RDP file, will read and apply those fields to the connection being imported.

Tested by verifying that the properties are reflected correctly in the UI (Config panel) and influence the behavior of the connection as expected.

asherbiny committed:
Merge branch 'v1.78.2-dev' into v1.78.2-dev

Kvarkas commented:
@asherbiny hello, thanks for PR! I ask copilot to review a changes and please check if you happy with suggestions will aprove the merge

asherbiny committed:
Add comment for duplicate cases
Cases 3 & 4 both require the user to manually enter Gateway credentials. Comment added for code readability.

asherbiny committed:
Merge branch 'v1.78.2-dev' into v1.78.2-dev

asherbiny commented:
@Kvarkas I've added a comment to explain the duplicate cases.
As for adding a 'default' statement, I think it's better to leave it off to keep consistency with the rest of the code in that file. Without a default, the property will be default-initialized to the first enum value, which is the desired behavior.
